cfpb/cfgov-refresh
# Modifying front-end process to use Sauce Tunnel

Modifying front-end process to use Sauce Tunnel

## Changes

- Modified `config/environment.js ` to add `SAUCE_TUNNEL` and default value. 
- Modified `gulp/tasks/test.js` to create a Sauce tunnel workflow and handle command line params. 
- Modified `npm-shrinkwrap.json` to remove unnecessary dependencies. 
- Modified `package.json` to add `sauce-tunnel`.
- Modified `test/browser_tests/conf.js` to add a `_useSauceLabs` method. This also changes the way the default sauce flag works. You only run on sauce when passing the flag, which is more aligned with the expected testing behavior. 
- Modified `test/browser_tests/default-suites.js` to remove IE8 and switch to Windows 7. 
- Modified `tox.ini` to use the appropriate variables. 

## Testing

1. Run `npm install`.
2. Run `gulp build`. 
3. Run `gulp test:acceptance --fast` and verify that the tests should run as normal.
4. Set your Sauce credentials in the `.env` file.
5. Run `gulp test:acceptance --fast --sauce`  and observe the log as the tunnel is created and the tests start.  Login to sauce labs and verify that the test results match those viewed locally.

## Screenshots

- 
<img width="1177" alt="screen shot 2017-06-15 at 3 28 08 pm" src="https://user-images.githubusercontent.com/1696212/27198217-448b8f72-51df-11e7-95f1-75f07c92ebff.png">


## Notes

- The Sauce Tunnel package contains binaries which are 40MB. This is larger than most of our other NPM packages. It might make sense to switch to a different package which lazy loads the binaries on first usage. 


## Checklist
